blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodCounterRollupTest.java

```diff
@@ -1,93 +1,758 @@
+        //          | <-- one time unit --> |                       |
+        //    ------+-----------------------+-----------------------+-----
+        //          1                       1                       1
+        //          2                       2                       2
+        //          3                       3                       3
+        //          4                       5                       6
+        //          L                       L                       L
+        //          |                                               |
+        //          | <-------------- two time units -------------> |
+        //                             (3+4+5)/2=6
+        //
+        //
+        // Samples As Intervals:
+        //                                                           _______________________
+        //                                   _______________________|          5            |
+        //           _______________________|          4            |                       |
+        //          |          3            |                       |                       |
+        //          |                       |                       |                       |
+        //          |                       |                       |                       |
+        //    ------+-----------------------+-----------------------+-----------------------+-----
+        //          1                       1                       1                       1
+        //          2                       2                       2                       2
+        //          3                       3                       3                       3
+        //          4                       5                       6                       7
+        //          L                       L                       L                       L
+        //          |                                                                       |
+        //          | <------------------------- three time units ------------------------> |
+        //                                         (3+4+5)/3=4
+        //
+
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        input.add(new Points.Point<SimpleNumber>(1234L, new SimpleNumber(3L)));
+        input.add(new Points.Point<SimpleNumber>(1235L, new SimpleNumber(4L)));
+        input.add(new Points.Point<SimpleNumber>(1236L, new SimpleNumber(5L)));
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(12L, count.longValue());
+        assertEquals(6.0d, rollup.getRate(), 0.00001d); // (3+4+5)/(3-1)=6
+        assertEquals(3, rollup.getSampleCount());
```

Comment:
@VinnyQ I don't think that's correct. It's more like an average. If we want to determine change over time, we can calculate `(end value - start value) / (end time - start time)`, which would give us `(5-3)/(3-1)`, which is 1 percentage point per hour. That makes sense to me for a variable named "`rate`", given the data. However, rollups calculate things like averages, not rates of change. That's what we return from the query service; that's what we see on our grafana boards. The average of those three samples is `(3+4+5)/3=4`, which seems right to me for an average.

The numerator is what we'd expect of an average. The denominator is what we'd expect of a delta. It's unclear to me what it's supposed to be. =/

I just realized: I think what it's trying to calculate, specifically, is "average value per time unit", rather than "average of all samples". But if that's the case, then there's an additional problem in that missing samples (i.e. time units for which there is no value recorded) are effectively treated as zero, bringing the whole average down.
If our time unit is one hour